src/api/middleware/validators/company.js

```diff
@@ -77,3 +78,30 @@ export const getOffers = useExpressValidators([
         .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
         .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND)
 ]);
+
+export const checkConcurrent = useExpressValidators([
+    existingCompanyParamValidator,
+
+    body("publishDate", ValidationReasons.DEFAULT)
+        .optional()
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
+        .customSanitizer(normalizeDate),
+
+    body("publishEndDate", ValidationReasons.DEFAULT)
+        .optional()
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
+        .customSanitizer(normalizeDate)
+        .custom((publishEndDate, { req }) => publishEndDate > req.body.publishDate)
+        .withMessage(ValidationReasons.MUST_BE_AFTER("publishDate")),
+]);
+
+export const setDefaultValuesConcurrent = (req, res, next) => {
+    if (!req.body?.publishDate) req.body.publishDate = new Date(Date.now()).toISOString();
+
+    if (!req.body?.publishEndDate) {
+        const publishDateMs = Date.parse(req.body.publishDate);
```

Comment:
```suggestion
        const publishDateMS = Date.parse(req.body.publishDate);
```

Reply:
"If validation only happens after this line, there's no guarantee you'll receive valid dates, which will make this function crash (#191)"

This will fail if the date is a malformed string, for example